### PR TITLE
Collector Config

### PR DESCRIPTION
**Pull request by anushree-n**
Create a config : Includes endpoint and metrics metadata
Extract config information
Create nginx collector using the config

### CODE REVIEW

**Comment by anushree-n** [comment]
I signed it.

File: collector/collector.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
```
Comment (vmarmol) [inline collector/collector.go]: rename file to `generic_collector.go`?

File: collector/collector.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+type GenericCollector struct {
+	//name of the collector
+	name string
+
+	//holds information extracted from the config file for a collector
+	configFile Config
+}
+
+type Config struct {
```
Comment (vmarmol) [inline collector/collector.go]: Move this to a `config.go`. This part is general, this collector is one of possibly many implementations.

File: collector/collector.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	//the endpoint to hit to scrape metrics
+	Endpoint string `json:"endpoint"`
+
+	//holds information about different metrics that can be collected
+	MetricsConfig []MetricConfig `json:"metricsConfig"`
```
Comment (vmarmol) [inline collector/collector.go]: nit: we tend to use_naming_with_underscores for JSON fields insteadOfCamelcase. So `metrics_config` and similarly elsewhere.

File: collector/collector.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	//the name of the metric
+	Name string `json:"name"`
+
+	//enum type for the metric type
+	MetricType MetricType `json:"metricType"`
+
+	//data type of the metric (eg: integer, string)
+	Units string `json:"units"`
+
+	//the frequency at which the metric should be collected (in seconds)
```
Comment (vmarmol) [inline collector/collector.go]: nit: remove "in seconds" since it is a duration

File: collector/collector.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+		name: collectorName, configFile: configInJSON,
+	}, nil
+}
+
+//Returns name of the collector
+func (collector *GenericCollector) Name() string {
+	return collector.name
+}
+
+//Returns the next collection time and collected metrics for the collector; Returns the next collection time and an error message in case of any error during metrics collection
```
Comment (vmarmol) [inline collector/collector.go]: reword?

File: collector/collector_test.go [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+package collector
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConfig(t *testing.T) {
```
Comment (vmarmol) [inline collector/collector_test.go]: Maybe add a failure to parse test?

**Comment by vmarmol** [comment]
Looks good overall! Minor comments

**Comment by vmarmol** [comment]
Let me know when this is ready for another look @anushree-n :)

**Comment by anushree-n** [comment]
I have made the changes you suggested, @vmarmol :)

File: collector/generic_collector_test.go [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+)
+
+func TestConfigWithErrors(t *testing.T) {
+	assert := assert.New(t)
+
+	invalidJSON := "{"
```
Comment (vmarmol) [inline collector/generic_collector_test.go]: You should be able to use a raw string for this:

http://play.golang.org/p/9xLuph6MEQ

Makes it more readable :)

Comment (anushree-n) [inline collector/generic_collector_test.go]: Done. :)

**Comment by vmarmol** [comment]
LGTM, thanks @anushree-n! Can we squash commits? and we'll merge.

**Pull request merged**